tests/dashboard/alerts.py

```diff
@@ -0,0 +1,137 @@
+from time import sleep
+
+from ceph.ceph_admin import CephAdmin
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def execute_commands(node, commands):
+    for cmd in commands:
+        out, err = node.exec_command(sudo=True, cmd=cmd)
+        log.info(f"Output:\n {out}")
+        log.error(f"Error:\n {err}")
+
+
+def run(ceph_cluster, **kw):
+    """
+    Prepares the cluster & runs alerts test Scenarios.
+    Args:
+        ceph_cluster (ceph.ceph.Ceph): ceph cluster
+        kw: Args that need to be passed to the test for initialization
+    Returns:
+        1 -> Fail, 0 -> Pass
+    """
+    log.info(run.__doc__)
+    config = kw["config"]
+    alert = config.get("alert")
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    if alert == "HEALTH_ERR":
+        if not generate_health_error_alert(alert, cephadm):
+            return 1
+    elif alert == "HEALTH_WARN":
+        if not generate_health_warn_alert(alert, cephadm):
+            return 1
+    return 0
+
+
+def validate_alerts(alert: str, node: CephAdmin):
+    """
+    Method to validate genearted alerts
+    Args:
+        alert: name of the alert
+        node: name of the cephadmin node
+    Returns: True -> pass, False -> failure
+    """
+    cmd = "ceph health detail"
+    out, err = node.shell([cmd])
+    if alert in out:
+        log.info(f"alert {alert} is generated successfully")
+        return True
+    else:
+        raise Exception(f"alert {alert} is not generated.")
+        return False
+
+
+def generate_health_error_alert(alert: str, node: CephAdmin) -> bool:
+    """
+    Method to generate health error alert
+    Args:
+        alert: name of the alert to be generated
+        node: name of the installer node
+    Returns: True -> pass, False -> failure
+
+    """
+    cmd = "ceph osd set-full-ratio .25"
+    try:
+        out, err = node.shell([cmd])
+        log.debug("changed the osd set-full-ratio")
+    except Exception:
+        log.error("Failed to change the osd set-full-ratio")
+        log.error(traceback.format_exc())
+        return False
+
+    if not validate_alerts(alert, node):
+        log.error(f"alert : {alert} has failed")
+        return False
+    # Resetting to default
+    cmd = "ceph osd set-full-ratio .95"
+    try:
+        out, err = node.shell([cmd])
+        log.debug("changed the osd set-full-ratio")
+    except Exception:
+        log.error("Failed to change the osd set-full-ratio")
+        log.error(traceback.format_exc())
+        return False
+    return True
+
+
+def generate_health_warn_alert(alert: str, admin_node: CephAdmin) -> bool:
+    """
+    Method to generate health warn alert
+    Args:
+        alert: name of the alert to be generated
+        node: name of the installer node
+    Returns: True -> pass, False -> failure
+
+    """
+    out, _ = admin_node.shell(
+        args=["ceph", "orch", "ps", "--daemon_type", "mon", "-f", "json"]
+    )
+    daemon_obj = loads(out)
+
+    if daemon_obj:
```

Comment:
This method can be used to get the host object for the specified daemon, after which , you can use the same host obj to execute commands on that node.  In your case, a specified mon daemon.

https://github.com/red-hat-storage/cephci/blob/62e2f2db91310c48673cecb87ee25303f3775f1f/ceph/rados/core_workflows.py#L894

Reply:
Thanks @pdhiran for that reference. 
But in that function it returns first host suppose we have 3 mon host,
But In this code I am using last mon host for stop operation.

Reply:
You can pass the name of the mon daemon to get it's placement. If you do not pass the mon name, and just pass daemon_type as mon, only then it would give you the 1st mon daemon it finds on the cluster.

Reply:
So once again i need to add code to find mon daemon Hence I feel the existing code covers both.